File: 1. Two Sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int len = nums.length;
        Map<Integer, Integer> map = new HashMap<>();
        
        for (int i = 0; i < len; i++) {
            if (map.containsKey(target - nums[i])) {
                return new int[]{map.get(target - nums[i]), i};
            } else {
                map.put(nums[i], i);
            }
        }
        return new int[]{-1, -1};
    }
}

File: Go/1. Two Sum.go
func twoSum(nums []int, target int) []int {
    num_idx := make(map[int]int)
    for i, num := range nums {
        if idx, contains := num_idx[target - num]; contains {
            return []int{i, idx}
        }
        num_idx[num] = i
    }
    return nil
}
